numba/np/arrayobj.py

```diff
@@ -4331,6 +4331,28 @@ def array_ascontiguousarray(context, builder, sig, args):
     return _as_layout_array(context, builder, sig, args, output_layout='C')
 
 
+def _array_ascontiguous_scalar(a):
```

Comment:
I'm not too familiar with the Fortran array layout (I never use it), but I suspect reusing the same function here is not completely correct because although it shouldn't be meaningful, the types won't accurately reflect the desired layout.

In particular based on existing code, this will always give the `layout='C'`, which won't be correct for the Fortran function. While this probably isn't meaningful in this case, I'm not sure what impact it might have when mixed with other APIs.

I'm not sure if this can be done with the higher level APIs, but I'll defer to people who are more knowledgable (perhaps @stuartarchibald).

Reply:
For single element arrays, I think they should be the same. I printed `np.ascontiguousarray(1).flags`. Both  `C_CONTIGUOUS ` and `F_CONTIGUOUS` were `True`. Same flags apart from `OWNDATA` flag were returned by this implementation. This also applies for `asfortranarray`.